frontend/src/Explorer/View/Blocks.purs

```diff
@@ -59,11 +57,19 @@ blocksView state =
                             <> " / " <>
                             (translate (I18nL.common <<< I18nL.cSlot) lang')
                       ]
-                , case state ^. latestBlocks of
+                , case state ^. currentBlocksResult of
                       NotAsked  -> emptyBlocksView ""
                       Loading   -> emptyBlocksView $ translate (I18nL.common <<< I18nL.cLoading) lang'
-                      Failure _ -> emptyBlocksView $ translate (I18nL.block <<< I18nL.blNotFound) lang'
+                      Failure _ -> failureView lang'
                       Success blocks ->
+                          let paginationViewProps =
+                                  { label: translate (I18nL.common <<< I18nL.cOf) $ lang'
+                                  , currentPage: state ^. (viewStates <<< blocksViewState <<< blsViewPagination)
+                                  , maxPage: (+) minPagination <<< flip (/) maxBlockRows $ length blocks
```

Comment:
Simpler is `(minPagination / maxBlockRows) + (length blocks)` if that's the goal.